diff --git a/glance.pp b/glance.pp
index 98c89e70..bcbf3132 100644
--- a/glance.pp
+++ b/glance.pp
@@ -1,5 +1,5 @@
 class openstack::glance::params (
-  $service_enabled = true,
+  $service_enabled = false,
   $api_port = 9292,
   $api_host,
   $region_name = undef,
@@ -14,6 +14,7 @@ class openstack::glance::params (
   $glance_delete_interval = 6,
   $rbd_store_pool = 'images',
   $rbd_store_ceph_conf = '/etc/ceph/ceph.conf',
+  $remote_host,
 ) { }
 
 
@@ -129,7 +130,16 @@ class openstack::glance::haproxy
     server_name => 's-glance',
     public_port => $api_port,
     private_port => $api_port,
-    private_ip_address => $api_host,
+    public_ip_address => "0.0.0.0"
+    private_ip_address => $remote_host,
+  }
+
+  platform::haproxy::proxy { 'glance-registry':
+    server_name => 's-glance-registry',
+    public_port => 9191,
+    private_port => 9191,
+    public_ip_address => "0.0.0.0"
+    private_ip_address => $remote_host,
   }
 }
 
